src/main/cms/cms.c

```diff
@@ -1026,72 +1025,77 @@ void cmsUpdate(uint32_t currentTimeUs)
         // Scan 'key' first
         //
 
-        uint8_t key = KEY_NONE;
-
-        if (IS_MID(THROTTLE) && IS_LO(YAW) && IS_HI(PITCH) && !ARMING_FLAG(ARMED)) {
-            key = KEY_MENU;
-        }
-        else if (IS_HI(PITCH)) {
-            key = KEY_UP;
-        }
-        else if (IS_LO(PITCH)) {
-            key = KEY_DOWN;
-        }
-        else if (IS_LO(ROLL)) {
-            key = KEY_LEFT;
-        }
-        else if (IS_HI(ROLL)) {
-            key = KEY_RIGHT;
-        }
-        else if (IS_HI(YAW) || IS_LO(YAW))
-        {
-            key = KEY_ESC;
-        }
+        uint8_t key = CMS_KEY_NONE;
 
-        if (key == KEY_NONE) {
-            // No 'key' pressed, reset repeat control
-            holdCount = 1;
-            repeatCount = 1;
-            repeatBase = 0;
+        if (externKey != CMS_KEY_NONE) {
```

Comment:
IMO current stick-key handling code should be separated into own function(s) (cmsProcessStickKey? cmsUpdateStickKey? )
and then 
```
  if (externKey != CMS_KEY_NONE) {
    delay = cmsHandleKey(pCurrentDisplay, externKey);
    cmsProcessStickKeyReset(delay);
    externKey = CMS_KEY_NONE;
  } else {
   cmsProcessStickKey();
  }

```

current function is getting too complicated (it would be ideal to separate stick-key mapping from repeat)

Reply:
Yes, but what should a cmsProcessStickKeyReset do?

Reply:
it should probably be  `cmsProcessStickKeyStop` - stop autorepeat, do nothing until stick position is CMS_KEY_NONE.
And set `rcDelayMs` if sticks are centered ...

Reply:
I agree, but I think this more to do with refactoring the CMS, and not with HoTT Textmode.

Reply:
Yes, but these problems were quite localized and hidden until this PR ...
The code was far from ideal, this PR is exposing its design flaws